### PR TITLE
[i18n/audio] Install UiStringContextProvider in voter-facing apps

### PR DESCRIPTION
**Pull request by kofi-q**
## Overview

_Related spec [here](https://docs.google.com/document/d/1a2GRj1g4aOawEssPdTCgbv-qs28d-M_dyMEfzm4I4nM/edit?usp=sharing)._

Installing the shared `UiStringContextProvider` in `Vx[Scan|Mark|Mark-Scan]` to connect `<UiString>` instances to the backend API.

Only enabling the context provider for dev/prod app instances (and integration tests) and leaving it disabled for the functional jest tests, since it was going to be too much unnecessary overhead to add mock versions of the UiString API endpoints and set up expectations in all existing tests that render the root `<App>` component directly.

## Testing Plan
- Unit test coverage for the added conditional provider installation
- Planning on following up later integration test coverage to verify e2e functionality from configuration through changing languages, once all the pieces are ready.

## Checklist

- ~[ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.~

- ~[ ] I have added a screenshot and/or video to this PR to demo the change~
- ~[ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates~

### CODE REVIEW

File: apps/mark-scan/frontend/src/app.tsx [hunk 1]
```diff
@@ -149,14 +159,19 @@ export function App({
...
-                  hardware={hardware}
-                  storage={storage}
-                  screenReader={screenReader}
-                  reload={reload}
-                  logger={logger}
-                />
-                <SessionTimeLimitTracker />
+                <UiStringsContextProvider
+                  api={uiStringsApi}
+                  disabled={!enableStringTranslation}
```
Comment (kofi-q) [inline apps/mark-scan/frontend/src/app.tsx:164]: The positive-to-negative mapping here feels admittedly awkward - initially had a `disableStringTranslationForTesting` prop on the `<App>` component, but that would've meant requiring every test using `<App>` to remember to explicitly disable, vs explicitly enabling once in `index.tsx`.

Comment (arsalansufi) [reply apps/mark-scan/frontend/src/app.tsx:164]: Ah yes the art of enable/disable params haha 🎨

After taking a sec to reason through the various possible permutations of params on `App` and `UiStringsContextProvider`, I like what you've landed on

File: libs/ui/src/ui_strings/ui_strings_context.tsx [hunk 2]
```diff
@@ -7,13 +7,18 @@ import { AudioContextProvider } from './audio_context';
...
 }
 
 export function UiStringsContextProvider(
   props: UiStringsContextProviderProps
 ): JSX.Element {
-  const { api, children, noAudio } = props;
+  const { api, children, disabled, noAudio } = props;
+
+  if (disabled) {
+    return <React.Fragment>{children}</React.Fragment>;
```
Comment (arsalansufi) [inline libs/ui/src/ui_strings/ui_strings_context.tsx:20]: Could this be simplified down to `return children;`?

**Review by arsalansufi — APPROVED** [review]
LGTM!

File: libs/ui/src/ui_strings/ui_strings_context.tsx (hunk 2 above)
Comment (kofi-q) [reply libs/ui/src/ui_strings/ui_strings_context.tsx:20]: Oh yeah, good call!

**Pull request merged**